pkg/client/attach.go

```diff
@@ -170,35 +171,42 @@ func (c *ConmonClient) attach(ctx context.Context, cfg *AttachConfig) (err error
 		return nil
 	}
 
-	receiveStdoutError, stdinDone := c.setupStdioChannels(cfg, conn)
+	id := uuid.NewString()
+
+	receiveStdoutError, stdinDone := c.setupStdioChannels(cfg, conn, id)
 	if cfg.PostAttachFunc != nil {
 		if err := cfg.PostAttachFunc(); err != nil {
 			return fmt.Errorf("run post attach func: %w", err)
 		}
 	}
 
-	if err := c.readStdio(cfg, conn, receiveStdoutError, stdinDone); err != nil {
+	if err := c.readStdio(cfg, conn, id, receiveStdoutError, stdinDone); err != nil {
 		return fmt.Errorf("read stdio: %w", err)
 	}
 
 	return nil
 }
 
 func (c *ConmonClient) setupStdioChannels(
-	cfg *AttachConfig, conn *net.UnixConn,
+	cfg *AttachConfig, conn *net.UnixConn, id string,
 ) (receiveStdoutError, stdinDone chan error) {
 	receiveStdoutError = make(chan error)
 	go func() {
 		receiveStdoutError <- c.redirectResponseToOutputStreams(cfg, conn)
 	}()
 
 	stdinDone = make(chan error)
+	stdin := cfg.Streams.Stdin
+
 	go func() {
-		var err error
-		if cfg.Streams.Stdin != nil {
-			_, err = util.CopyDetachable(conn, cfg.Streams.Stdin, cfg.DetachKeys)
+		if stdin != nil {
+			c.attachReaders.Store(id, stdin)
```

Comment:
nit... We should reverse the logic and return if stdin == nil, and remove the nesting.

Reply:
Done, changed the logic to not even start the goroutine if `stdin` is `nil`.